src/components/forms/NetworkDevicesForm/edit/NetworkDevicePanel.tsx

```diff
@@ -0,0 +1,327 @@
+import {
+  ActionButton,
+  Button,
+  Form,
+  Input,
+  ScrollableContainer,
+  SidePanel,
+  useNotify,
+} from "@canonical/react-components";
+import type { ReactNode } from "react";
+import { type FC } from "react";
+import { useFormik } from "formik";
+import * as Yup from "yup";
+import type { LxdNicDevice } from "types/device";
+import NotificationRow from "components/NotificationRow";
+import NetworkDeviceIPAddressEdit from "components/forms/NetworkDevicesForm/edit/NetworkDeviceIPAddressEdit";
+import { useNetworks } from "context/useNetworks";
+import { useProfiles } from "context/useProfiles";
+import NetworkSelector from "pages/projects/forms/NetworkSelector";
+import NetworkAclSelector from "pages/networks/forms/NetworkAclSelector";
+import { getDeviceAcls, getExistingDeviceNames } from "util/devices";
+import type { FormDevice, FormNetworkDevice } from "util/formDevices";
+import { deduplicateName } from "util/formDevices";
+import { getInheritedNetworks } from "util/configInheritance";
+import {
+  supportsNicDeviceAcls,
+  getNetworkAcls,
+  combineAcls,
+} from "util/networks";
+import usePanelParams, { panels } from "util/usePanelParams";
+import type { InstanceAndProfileFormikProps } from "components/forms/instanceAndProfileFormValues";
+
+export interface NetworkDeviceFormValues {
+  name: string;
+  network: string;
+  acls?: string;
+  ipv4?: string;
+  ipv6?: string;
+}
+
+interface Props {
+  project: string;
+  formik: InstanceAndProfileFormikProps;
+  onSave?: () => void;
+}
+
+const NetworkDevicePanel: FC<Props> = ({
+  project,
+  formik: parentFormik,
+  onSave,
+}) => {
+  const notify = useNotify();
+  const panelParams = usePanelParams();
+  const { data: profiles = [] } = useProfiles(project);
+  const { data: networks = [] } = useNetworks(project);
+
+  const managedNetworks = networks.filter((network) => network.managed);
+
+  const existingDeviceNames = getExistingDeviceNames(
+    parentFormik.values,
+    profiles,
+  );
+  const inheritedNetworks = getInheritedNetworks(parentFormik.values, profiles);
+  const isInstance = parentFormik.values.entityType === "instance";
+  const deviceName = panelParams?.deviceName;
+  const device = parentFormik.values.devices.find(
+    (d) => d.name === deviceName,
+  ) as LxdNicDevice | undefined;
+
+  const inheritedDevice = inheritedNetworks.find((d) => d.key === deviceName);
```

Comment:
```suggestion
  const isInstance = parentFormik.values.entityType === "instance";

  const deviceName = panelParams?.deviceName;
  const device = parentFormik.values.devices.find(
    (d) => d.name === deviceName,
  ) as LxdNicDevice | undefined;

  const inheritedNetworks = getInheritedNetworks(parentFormik.values, profiles);
  const inheritedDevice = inheritedNetworks.find((d) => d.key === deviceName);
```
A bit nicer grouping.